### DISCUSSION TITLE
Dynamically sizing the kmeans kernel

### DISCUSSION THREAD

**Discussion by GitRjc** [davisking/dlib#2812] [Q&A]
In the DLIB Kernel Kmeans Clustering example, the number of features in (sample_type) is statically set to 2. The comments says that this value can be set to zero, and then dynamically set at runtime. It's not clear how to do this as it also affects the radial_basis_kernel and kcentriod size. How to do kmeans on an arbitrary number of features?

**Comment by davisking** [comment]
See http://dlib.net/faq.html#HowdoIsetthesizeofamatrixatruntime
